python/pyarrow/tests/test_compute.py

```diff
@@ -3092,6 +3092,12 @@ def test_map_lookup():
     result_all = pa.array([[1], None, None, [5, 7], None],
                           type=pa.list_(pa.int32()))
 
+    assert pc.map_lookup(arr,
+                         'one', 'first') == result_first
+    assert pc.map_lookup(arr,
+                         'one', 'last') == result_last
+    assert pc.map_lookup(arr,
+                         'one', 'all') == result_all
```

Comment:
Adding just the first case should be sufficient to cover this (the value for occurrence shouldn't impact how the query_key is converted)